src/test/java/org/junitpioneer/jupiter/EnvironmentVariableExtensionTests.java

```diff
@@ -179,6 +195,225 @@ void methodLevelShouldNotClashWithClassLevel() {
 
 	}
 
+	@Nested
+	@DisplayName("used with Clear, Set and Restore")
+	@Execution(SAME_THREAD) // Uses instance state
+	@TestInstance(TestInstance.Lifecycle.PER_CLASS) // Uses instance state
+	@TestClassOrder(ClassOrderer.OrderAnnotation.class)
+	@WritesEnvironmentVariable // Many of these tests write, many also access
+	class CombinedClearSetRestoreTests {
+
+		Map<String, String> initialState;
+
+		@BeforeAll
+		public void beforeAll() {
+			HashMap<String, String> envVars = new HashMap<>();
+			envVars.putAll(System.getenv()); //detached
+
+			initialState = envVars;
```

Comment:
This would be a nice change, but its not quite the same.  `Map.copyOf` has this bit of logic:
```java
if (map instanceof ImmutableCollections.AbstractImmutableMap) {
            return (Map<K,V>)map;
}
```
So if the passed Map is already Immutable, it doesn't do anything further.  BUT, we are hacking with reflection, so we are actually modifying the unmodifiable.  Thus, I really do need to copy the env. vars. out to get a completely detached copy.